core/ledger/kvledger/txmgmt/statedb/statecouchdb/couchdb.go

```diff
@@ -232,73 +231,45 @@ func (dbclient *couchDatabase) createDatabaseIfNotExist() error {
+			return errors.Wrapf(err, "error parsing CouchDB URL: %s", dbclient.couchInstance.url())
 		}
 
-		logger.Debugf("[%s] Database already exists", dbclient.dbName)
-
-		logger.Debugf("[%s] Exiting CreateDatabaseIfNotExist()", dbclient.dbName)
-
-		return nil
-	}
-
-	logger.Debugf("[%s] Database does not exist.", dbclient.dbName)
-
-	connectURL, err := url.Parse(dbclient.couchInstance.url())
-	if err != nil {
-		logger.Errorf("URL parse error: %s", err)
-		return errors.Wrapf(err, "error parsing CouchDB URL: %s", dbclient.couchInstance.url())
-	}
-
-	//get the number of retries
-	maxRetries := dbclient.couchInstance.conf.MaxRetries
+		//get the number of retries
+		maxRetries := dbclient.couchInstance.conf.MaxRetries
 
-	//process the URL with a PUT, creates the database
-	resp, _, err := dbclient.handleRequest(http.MethodPut, "CreateDatabaseIfNotExist", connectURL, nil, "", "", maxRetries, true, nil)
-
-	if err != nil {
-
-		// Check to see if the database exists
-		// Even though handleRequest() returned an error, the
-		// database may have been created and a false error
-		// returned due to a timeout or race condition.
-		// Do a final check to see if the database really got created.
-		dbInfo, couchDBReturn, errDbInfo := dbclient.getDatabaseInfo()
-		//If there is no error, then the database exists,  return without an error
-		if errDbInfo == nil && dbInfo != nil && couchDBReturn.StatusCode == 200 {
-
-			errSecurity := dbclient.applyDatabasePermissions()
-			if errSecurity != nil {
-				return errSecurity
+		//process the URL with a PUT, creates the database
+		resp, _, err := dbclient.handleRequest(http.MethodPut, "CreateDatabaseIfNotExist", connectURL, nil, "", "", maxRetries, true, nil)
+		if err != nil {
+			// Check to see if the database exists
+			// Even though handleRequest() returned an error, the
+			// database may have been created and a false error
+			// returned due to a timeout or race condition.
+			// Do a final check to see if the database really got created.
+			dbInfo, couchDBReturn, dbInfoErr := dbclient.getDatabaseInfo()
+			//If there is no error, then the database exists,  return without an error
+			if dbInfoErr != nil || dbInfo == nil || couchDBReturn.StatusCode != 200 {
+				return err
 			}
-
-			logger.Infof("[%s] Created state database", dbclient.dbName)
-			logger.Debugf("[%s] Exiting CreateDatabaseIfNotExist()", dbclient.dbName)
-			return nil
 		}
-
-		return err
-
+		defer closeResponseBody(resp)
+		logger.Infof("Created state database %s", dbclient.dbName)
 	}
-	defer closeResponseBody(resp)
 
-	errSecurity := dbclient.applyDatabasePermissions()
-	if errSecurity != nil {
-		return errSecurity
+	if dbclient.dbName != "_users" {
+		errSecurity := dbclient.applyDatabasePermissions()
+		if errSecurity != nil {
+			return errSecurity
+		}
 	}
-
-	logger.Infof("Created state database %s", dbclient.dbName)
-
+	logger.Debugf("[%s] Database already exists", dbclient.dbName)
```

Comment:
I think this debug is misplaced... it will print this message even if the database gets created in this function.